src/index/index.c

```diff
@@ -51,33 +51,79 @@ void freeKey(SIValue *key) {
 }
 
 /* Construct key and retrieve index from Redis keyspace */
-Index* Index_Get(RedisModuleCtx *ctx, const char *graph, const char *label, const char *property) {
+Index* Index_Get(RedisModuleCtx *ctx, const char *graph, const char *label, char *property) {
   Index *idx = NULL;
-  // Open key with read-only access
-  RedisModuleKey *key = _index_LookupKey(ctx, graph, label, property, false);
 
-  if (RedisModule_ModuleTypeGetType(key) == IndexRedisModuleType) {
-    idx = RedisModule_ModuleTypeGetValue(key);
-  }
+  LabelStore *store = LabelStore_Get(ctx, STORE_NODE, graph, label);
+  if (!store) return NULL;
 
-  RedisModule_CloseKey(key);
+  size_t *index_id = TrieMap_Find(store->properties, property, strlen(property));
+
+  // index_id will be NULL if the label-property pair is not indexed
+  if (!index_id || index_id == TRIEMAP_NOTFOUND) return NULL;
 
-  return idx;
+  return gc->indices[*index_id];
 }
 
-int Index_Delete(RedisModuleCtx *ctx, const char *graphName, const char *label, const char *prop) {
+int Index_Delete(RedisModuleCtx *ctx, const char *graphName, Graph *g, const char *label, char *prop) {
+  LabelStore *store = LabelStore_Get(ctx, STORE_NODE, graphName, label);
+  assert(store);
+
+  size_t *index_id = TrieMap_Find(store->properties, prop, strlen(prop));
+
+  if (index_id == TRIEMAP_NOTFOUND) {
+    assert(0);
+  } else if (index_id == NULL) {
+    return INDEX_FAIL;
+  }
+
+  size_t dest_id = *index_id;
+
+  // TODO a lot of this should be unnecessary now
```

Comment:
By which I mean that we should move to accessing indices by ID from an array (such as within GraphContext).